sdk/appconfiguration/azure-data-appconfiguration/src/main/java/com/azure/data/appconfiguration/ConfigurationClient.java

```diff
@@ -670,7 +670,7 @@ public ConfigurationSetting setReadOnly(ConfigurationSetting setting, boolean is
     @ServiceMethod(returns = ReturnType.SINGLE)
     public Response<ConfigurationSetting> setReadOnlyWithResponse(ConfigurationSetting setting, boolean isReadOnly,
                                                                   Context context) {
-        return client.setReadOnly(setting, isReadOnly, context).block();
+        return client.setReadOnlyWithResponseSync(setting, isReadOnly, context);
     }
```

Comment:
List API's to be updated in a follow-up PR once https://github.com/Azure/azure-sdk-for-java/issues/30031 is resolved.